src/tilesource.js

```diff
@@ -217,12 +226,49 @@ $.TileSource.prototype = /** @lends OpenSeadragon.TileSource.prototype */{
      * Subclasses should override this if tileSizes can be different at different levels
      *   such as in IIIFTileSource.  Code should use this function rather than reading
      *   from .tileSize directly.  tileSize may be deprecated in a future release.
+     * @deprecated
```

Comment:
I actually like to just remove the documentation for deprecated items (along with adding a deprecation notice to the changelog).